### PR TITLE
feat: add filtering and early exit to `api compare` command

### PR DESCRIPTION
**Pull request by lemmih**
## Summary of changes

Changes introduced in this pull request:

- Add `--filter` flag limiting test runs to a subset of RPC methods.
- Add `--fast-fail` flag that cancels a test run on the first failure.
- Sort tests by RPC method name before running them.

## Reference issue to close (if applicable)

Part of #3641

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by ruseinov — APPROVED** [review]

File: src/tool/subcommands/api_cmd.rs [hunk 1]
```diff
@@ -28,6 +28,12 @@ pub enum ApiCommands {
         /// Snapshot input paths. Supports `.car`, `.car.zst`, and `.forest.car.zst`.
         #[arg()]
         snapshot_files: Vec<PathBuf>,
+        /// Test filter
```
Comment (aatifsyed) [inline src/tool/subcommands/api_cmd.rs:31]: Could be more helpful - how do I, the user, use it?

File: src/tool/subcommands/api_cmd.rs [hunk 2]
```diff
@@ -28,6 +28,12 @@ pub enum ApiCommands {
...
+        #[arg(long, default_value = "")]
+        filter: String,
+        /// Cancel test run on the first failure
+        #[arg(long)]
+        fail_fast: bool,
```
Comment (aatifsyed) [inline src/tool/subcommands/api_cmd.rs:36]: I'd prefer `--no-fail-fast` to match `cargo test` and `cargo nextest`

**Review by aatifsyed — APPROVED** [review]

File: src/tool/subcommands/api_cmd.rs (hunk 1 above)
Comment (lemmih) [reply src/tool/subcommands/api_cmd.rs]: Described the flag in more detail.

File: src/tool/subcommands/api_cmd.rs (hunk 2 above)
Comment (lemmih) [reply src/tool/subcommands/api_cmd.rs:36]: Clap has little support for `--no-*` flags. Let's punt this for another time.

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**